WordPress/gutenberg
# Add option to skip PublishSidebar on publishing

This PR adds a new option to skip the PublishSidebar on publishing.

### How it works

The option is surfaced both in the "Tools" section within the general menu and the "Publish sidebar" itself:

![peek 2018-09-11 20-24](https://user-images.githubusercontent.com/583546/45379653-f5821b80-b600-11e8-8cf5-eb4364f1efa3.gif)

This is how it looks when the option is enabled:

![peek 2018-09-11 20-25](https://user-images.githubusercontent.com/583546/45379667-fd41c000-b600-11e8-90a5-ea0864140e8c.gif)

This is how it looks when the option is disabled:

![peek 2018-09-11 20-26](https://user-images.githubusercontent.com/583546/45379678-03d03780-b601-11e8-9d57-930aab0c42f8.gif)

### Testing

- Open a new post
- Add some content
- Click "Publish..." button (which should show three dots at the end).

The expected result is that the Pre-Publish panel is opened and a checkbox is shown at the bottom of it.

- Disable the checkbox, and cancel publishing.
- Click "Publish" button again (which shouldn't have the dots).

The expected result is that the post is published directly, without the pre-publish checks.

These can also be disabled from "Tools" section of the general sidebar. Play a bit with those checks and make sure it works as expected.

### TODO 

- [x] Add "Don't show this sidebar again" checkbox to the Publish Sidebar.
- [x] Better style the dismissable checkbox.
- [x] Move preference logic to `core/editor`.
